WordPress/gutenberg
# Adds 'nofollow' setting to Button block

## What?

Add the ability to toggle the `nofollow` for Button's link. 

Partial implementation of https://github.com/WordPress/gutenberg/issues/54091
Prior art for Rich text link https://github.com/WordPress/gutenberg/pull/53945

## Why?

Check https://github.com/WordPress/gutenberg/pull/53945#issue-1866939265

## How?

- Place the `nofollow` option in the `Advanced` settings area
- Update the [linkValue](https://github.com/WordPress/gutenberg/blob/0792221dd597290fe512b109a1469f2057f0dda4/packages/block-library/src/button/edit.js#L154) for persisting the Advanced settings state

## Testing Instructions




- Open Post Editor
- Add new Button block
- Add a link to the Button
- Click Edit link
- Toggle the `opensInNewTab` and `nofollow` 
- Check and confirm whether the `rel` and `target` attributes has been inserted correctly

| opensInNewWindow | nofollow | Generated                      |
| --------------------------- | --------------------- | ---------------------------------------- |
| ✅                         | ❌                   | `target="_blank" rel="noreferrer noopener"` |
| ❌                         | ✅                   | `target="" rel="nofollow"`               |
| ✅                         | ✅                   | `target="_blank" rel="noreferrer noopener nofollow"` |
| ❌                         | ❌                   | `target="" rel=""`                      |

### Testings
- Verify and update these e2e tests
   `npm run test:e2e:playwright -- test/e2e/specs/editor/blocks/links.spec.js`
   `npm run test:e2e:playwright -- test/e2e/specs/editor/blocks/buttons.spec.js`

- Insert new unit testing for updateLinkAttributes
  `npm run test:unit -- packages/block-library/src/button`

### Testing Instructions for Keyboard


## Screenshots or screencast 

https://github.com/WordPress/gutenberg/assets/10071857/de3bcbc1-6bb7-45e8-b6eb-d7da7e6d974e

cc: @getdave
